fbw-a32nx/src/systems/tcas/src/components/TcasComputer.ts

```diff
@@ -7,1189 +7,1304 @@
+            ? TCAS.TVTHR[this.sensitivity.getVar()]
+            : TCAS.TAU[this.sensitivity.getVar()][TaRaIndex.RA]) ||
+        Math.abs(traffic.relativeAlt) < TCAS.ZTHR[this.sensitivity.getVar()][TaRaIndex.RA]
+      ) {
+        altTest = TaRaIntrusion.RA;
+      } else if (
+        traffic.vTau < TCAS.TAU[this.sensitivity.getVar()][TaRaIndex.TA] ||
+        Math.abs(traffic.relativeAlt) < TCAS.ZTHR[this.sensitivity.getVar()][TaRaIndex.TA]
+      ) {
+        altTest = TaRaIntrusion.TA;
+      } else if (Math.abs(traffic.relativeAlt) < 1200) {
+        altTest = TaRaIntrusion.PROXIMITY;
+      }
+
+      // Perform acceleration test
+      // TODO FIXME: Proper HMD based true-to-life filtering
+      if (Math.abs(traffic.closureAccel) <= TCAS.ACCEL[this.sensitivity.getVar()][TaRaIndex.RA]) {
+        accelTest = TaRaIntrusion.RA;
+      } else if (Math.abs(traffic.closureAccel) <= TCAS.ACCEL[this.sensitivity.getVar()][TaRaIndex.TA]) {
+        accelTest = TaRaIntrusion.TA;
+      } else {
+        accelTest = TaRaIntrusion.PROXIMITY;
+      }
+
+      const desiredIntrusionLevel: TaRaIntrusion = Math.min(rangeTest, altTest, accelTest);
+      switch (traffic.intrusionLevel) {
+        case TaRaIntrusion.RA:
+          if (
+            this.activeRa.info === null ||
+            this.activeRa.secondsSinceStart < TCAS.MIN_RA_DURATION ||
+            (traffic.taTau > TCAS.TAU[this.sensitivity.getVar()][TaRaIndex.TA] * TCAS.VOL_BOOST &&
+              traffic.slantDistance > TCAS.DMOD[this.sensitivity.getVar()][TaRaIndex.TA] * TCAS.VOL_BOOST) ||
+            (traffic.vTau > TCAS.TAU[this.sensitivity.getVar()][TaRaIndex.TA] * TCAS.VOL_BOOST &&
+              Math.abs(traffic.relativeAlt) > TCAS.ZTHR[this.sensitivity.getVar()][TaRaIndex.TA] * TCAS.VOL_BOOST) ||
+            traffic.closureRate < TCAS.CLOSURE_RATE_THRESH
+          ) {
+            traffic.intrusionLevel = desiredIntrusionLevel;
+          }
+          break;
+        case TaRaIntrusion.TA:
+          switch (desiredIntrusionLevel) {
+            case TaRaIntrusion.RA:
+              if (!this.isSlewActive) {
+                traffic.intrusionLevel = desiredIntrusionLevel;
+                if (this.debug) console.log(`${traffic.ID} new intrusion level is ${desiredIntrusionLevel}`);
+              }
+              break;
+            case TaRaIntrusion.TA:
+              if (this.debug) console.log(`${traffic.ID} resetting seconds since last TA to 0`);
+              traffic.secondsSinceLastTa = 0;
+              break;
+            default:
+              if (traffic.secondsSinceLastTa >= TCAS.TA_EXPIRATION_DELAY) {
+                traffic.secondsSinceLastTa = 0;
+                traffic.intrusionLevel = desiredIntrusionLevel;
+                if (this.debug)
+                  console.log(
+                    `${traffic.ID} resetting seconds since last TA to ${traffic.secondsSinceLastTa} - new intrusion level is ${desiredIntrusionLevel}`,
+                  );
```

Comment:
```suggestion
                if (this.debug) {
                  console.log(
                    `${traffic.ID} resetting seconds since last TA to ${traffic.secondsSinceLastTa} - new intrusion level is ${desiredIntrusionLevel}`,
                  );
                }
```